File: src/main/java/com/mark/student/Student.java
package com.mark.student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest() {
        return (english > math) ? english : math;
    }

    public void print() {
        int average = getAverage();

        System.out.println(name + "\t" + english + "\t" + math + "\t" + getAverage() + "\t" + ((getAverage() >= 60) ? "PASS" : "FAILED"));
        char grading = 'F';
        switch (average/10) {
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
                default:
        }
        System.out.println(grading);
//        if (getAverage() >= 60) {
//            System.out.println("\tPass");
//        } else {
//            System.out.println("\tFailed");
//        }
    }

    public int getAverage() {
        return (english+math)/2;
    }
}

File: src/main/kotlin/com/mark/student/Student.kt
package com.mark.student

import java.util.*
import kotlin.math.max

fun main(args: Array<String>) {
//    userInput()
    val stu = Student("Mark", 77, 99)
    stu.print()
    print("High score: ${stu.highest()}" )
}

class Student(var name: String, var english: Int, var math: Int) {
    fun print() {
//        print(name + "\t" + english + "\t" + math + "\t" + getAverage() + "\t" + passOrFailed())
//        println("\t" + grading())
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")
    }

    fun grading() = when(getAverage()) {
            in 90..100 -> 'A'
            in 80..89 -> 'B'
            in 70..79 -> 'C'
            in 60..69 -> 'D'
            else -> 'F'
    }

    fun passOrFailed() = if (getAverage() >= 60) "PASS" else "FAILED"

    fun getAverage() =  (english+math)/2

    fun highest() = if (english > math) english else math
}

private fun userInput() {
    val scanner = Scanner(System.`in`)
    print("Please enter student's name: ")
    val name = scanner.next()
    print("Please enter student's english score: ")
    val english = scanner.nextInt()
    print("Please enter student's math score: ")
    val math = scanner.nextInt()
    val stu = Student(name, english, math)
    stu.print()
}
